### PR TITLE
Fix PReLU (Resolves #3466)

### LINKED ISSUE #3466: Parameteric ReLU only applied when training?
In Forward method of Paramteric ReLU, there is the following _if condition_ in line 101: ([link](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/parametric_relu_impl.hpp#L101))

```
 96 template<typename MatType>
 97 void PReLUType<MatType>::Forward(
 98     const MatType& input, MatType& output)
 99 {
100   output = input;
101   if (this->training)
102   {
103     #pragma omp for
104     for (size_t i = 0; i < input.n_elem; ++i)
105       output(i) *= (input(i) >= 0) ? 1 : alpha(0);
106   }
107 }
```
So the parametric relu is applied only when training the model and it works like a identity layer during inference/prediction. Is this the expected behaviour of PReLU?

### PR DESCRIPTION
**Pull request by mrdaybird**
Fixes #3466

- [x]  Fix PReLU
- [x]  Add tests

### CODE REVIEW

**Comment by mrdaybird** [comment]
@rcurtin I have added the tests as well. If you have any suggestions, do share.

File: src/mlpack/tests/ann/layer/parametric_relu.cpp [hunk 1]
```diff
@@ -94,3 +93,41 @@ TEST_CASE("PReLUGRADIENTTest", "[ANNLayerTest]")
...
+
+  FFN<L1Loss> model;
+  model.Add<Linear>(10);
+  model.Add<PReLU>(0.01);
+  model.Add<Linear>(3);
+  model.Add<PReLU>(0.01);
+  model.Add<Linear>(1);
+    
+  int epochs = 50;
+  ens::RMSProp optimizer(0.01, 32, 0.99, 1e-8, epochs*trainData.n_cols);
```
Comment (rcurtin) [inline src/mlpack/tests/ann/layer/parametric_relu.cpp:120]:
```suggestion
  ens::RMSProp optimizer(0.01, 32, 0.99, 1e-8, epochs * trainData.n_cols);
```

File: src/mlpack/tests/ann/layer/parametric_relu.cpp [hunk 2]
```diff
@@ -94,3 +93,41 @@ TEST_CASE("PReLUGRADIENTTest", "[ANNLayerTest]")
...
+  model.Train(trainData, trainLabels, optimizer);
+
+  arma::mat predictions;
+  model.Predict(trainData, predictions);
+  double msreTrain = ComputeMSRE(predictions, trainLabels);
+  model.Predict(testData, predictions);
+  double msreTest = ComputeMSRE(predictions, testLabels);
+
+  double relativeMSRE = std::abs((msreTest-msreTrain)/msreTrain);
```
Comment (rcurtin) [inline src/mlpack/tests/ann/layer/parametric_relu.cpp:129]:
```suggestion
  double relativeMSRE = std::abs((msreTest - msreTrain) / msreTrain);
```

**Review by rcurtin — APPROVED** [review]
Nice, thanks for the fix. Nice integration test, by the way.

Want to add something to `HISTORY.md`? If not I'll handle it during merge.  All of my.comments are tiny style fixes. :+1:

**Review by zoq — APPROVED** [review]
Looks good to me.

**Pull request merged**